lib/cloudant.js

```diff
@@ -745,7 +745,9 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   var id = data[idName];
   if (id) {
     self.getCurrentRevision(model, id, function(err, rev) {
-      if (err) return cb(err);
+      if (err && err.statusCode !== 404) {
+        return cb(err);
+      }
```

Comment:
I understand, but if we want to preserve the idea of what `updateOrCreate` does from a LoopBack perspective, then we should use `updateAttributes` to attempt the update, and then handle the failure as appropriate, either by making the record, or returning the error up the callback chain.